compiler/noirc_frontend/src/parser/parser/parse_many.rs

```diff
@@ -0,0 +1,108 @@
+use crate::token::Token;
+
+use super::Parser;
+
+impl<'a> Parser<'a> {
+    /// Parses a list of items separated by a token, optionally ending when another token is found.
+    /// The given function `f` must parse an item. If no item is parsed, `f` must report an error
+    /// and return `None`.
```

Comment:
"`f` must parse an item" reads to me like this expects to parse 1 or more items instead of 0 or more which I believe is the intention